src/test/java/com/microsoft/sqlserver/jdbc/bvt/bvtTest.java

```diff
@@ -1,678 +1,587 @@
+            conn = new DBConnection(connectionString);
+            String colName = table1.getColumnName(7);
+            String value = DBResultSet.getRow(table1, 0).get(7).toString();
+
+            String query = "SELECT * from " + table1.getEscapedTableName() + " where [" + colName + "] = ? ";
+
+            pstmt = conn.prepareStatement(query);
+            pstmt.setObject(1, new BigDecimal(value));
+           
+            rs = pstmt.executeQuery();
+            rs.verify(table1);
+        }
+        finally {
+            terminateVariation();
+        }
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    // Verify resultset using ResultSetMetaData
+    ///////////////////////////////////////////////////////////////////
+    @Test
+    public void testResultSet() throws SQLException {
+
+        try {
+            conn = new DBConnection(connectionString);
+            stmt = conn.createStatement();
+
+            String query = "SELECT * FROM " + table1.getEscapedTableName();
+            rs = stmt.executeQuery(query);
+
+            // verify resultSet
+            rs.verify(table1);
+        }
+        finally {
+            terminateVariation();
+        }
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    // Verify resultset and close resultSet
+    ///////////////////////////////////////////////////////////////////
+    @Test
+    public void testResultSetAndClose() throws SQLException {
+
+        try {
+            conn = new DBConnection(connectionString);
+            stmt = conn.createStatement();
+
+            String query = "SELECT * FROM " + table1.getEscapedTableName();
+            rs = stmt.executeQuery(query);
+            rs.currentTable = table1;
+
+            rs.verify(table1);
+            stmt.close();
+        }
+        finally {
+            terminateVariation();
+        }
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    // Verify two concurrent resultsets from same connection,
+    // separate statements
+    ///////////////////////////////////////////////////////////////////
+    @Test
+    public void testTwoResultsetsDifferentStmt() throws SQLException {
+
+        DBStatement stmt1 = null;
+        DBStatement stmt2 = null;
+        DBResultSet rs1 = null;
+        DBResultSet rs2 = null;
+        try {
+            conn = new DBConnection(connectionString);
+            stmt1 = conn.createStatement();
+            stmt2 = conn.createStatement();
+
+            String query = "SELECT * FROM " + table1.getEscapedTableName();
+            rs1 = stmt1.executeQuery(query);
+            rs1.currentTable = table1;
+
+            String query2 = "SELECT * FROM " + table2.getEscapedTableName();
+            rs2 = stmt2.executeQuery(query2);
+            rs2.currentTable = table2;
+
+            // Interleave resultset calls
+            rs1.next();
+            rs1.verifyCurrentRow(table1);
```

Comment:
currentTable variable in DBResultSet can be made protected, it will be less confusing. in line #429 currentTable is set and the same value is passed again in  DBResultSet#verifyCurrentRow (line #437)